mcs/class/System/Mono.AppleTls/Certificate.cs

```diff
@@ -119,89 +82,56 @@ internal SecCertificate (X509CertificateImpl impl)
 				return;
 			}
 
-			using (NSData cert = NSData.FromArray (impl.GetRawCertData ())) {
+			using (CFData cert = CFData.FromData (impl.GetRawCertData ())) {
 				Initialize (cert);
 			}
 		}
-#endif
 
-		public SecCertificate (X509Certificate2 certificate)
-		{
-			if (certificate == null)
-				throw new ArgumentNullException ("certificate");
-
-#if NATIVE_APPLE_CERTIFICATE
-			handle = certificate.Impl.GetNativeAppleCertificate ();
-			if (handle != IntPtr.Zero) {
-				CFObject.CFRetain (handle);
-				return;
-			}
-#endif
-
-			using (NSData cert = NSData.FromArray (certificate.RawData)) {
-				Initialize (cert);
-			}
-		}
-
-		void Initialize (NSData data)
+		void Initialize (CFData data)
 		{
 			handle = SecCertificateCreateWithData (IntPtr.Zero, data.Handle);
 			if (handle == IntPtr.Zero)
 				throw new ArgumentException ("Not a valid DER-encoded X.509 certificate");
 		}
 
-		[DllImport (Constants.SecurityLibrary)]
+		[DllImport (AppleTlsContext.SecurityLibrary)]
 		extern static IntPtr SecCertificateCopySubjectSummary (IntPtr cert);
 
 		public string SubjectSummary {
 			get {
 				if (handle == IntPtr.Zero)
 					throw new ObjectDisposedException ("SecCertificate");
 				
-				return CFString.FetchString (SecCertificateCopySubjectSummary (handle), releaseHandle: true);
+				try {
+					return CFString.AsString (SecCertificateCopySubjectSummary (handle));
+				}
+				finally {
+					CFObject.CFRelease (handle);
```

Comment:
You're releasing the `SecCertificate`'s handle here, not the return value from `SecCertificateCopySubjectSummary` (which is what the original code did).